ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/time/WallClockTimeIT.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.sandbox.services.time
+
+import java.time.Instant
+
+import akka.stream.scaladsl.Sink
+import com.digitalasset.api.util.TimestampConversion.fromInstant
+import com.digitalasset.grpc.GrpcException
+import com.digitalasset.grpc.adapter.client.akka.ClientAdapter
+import com.digitalasset.ledger.api.testing.utils.SuiteResourceManagementAroundAll
+import com.digitalasset.ledger.api.v1.testing.time_service.{
+  GetTimeRequest,
+  SetTimeRequest,
+  TimeServiceGrpc
+}
+import com.digitalasset.platform.sandbox.config.SandboxConfig
+import com.digitalasset.platform.sandbox.services.SandboxFixture
+import com.digitalasset.platform.services.time.TimeProviderType
+import org.scalatest.concurrent.{AsyncTimeLimitedTests, ScalaFutures}
+import org.scalatest.time.Span
+import org.scalatest.time.SpanSugar.convertIntToGrainOfTime
+import org.scalatest.{AsyncWordSpec, Matchers}
+import scalaz.syntax.tag.ToTagOps
+
+final class WallClockTimeIT
+    extends AsyncWordSpec
+    with SandboxFixture
+    with SuiteResourceManagementAroundAll
+    with AsyncTimeLimitedTests
+    with ScalaFutures
+    with Matchers {
+
+  override val timeLimit: Span = 15.seconds
+
+  override protected def config: SandboxConfig = SandboxConfig.default.copy(
+    timeProviderType = TimeProviderType.WallClock,
+  )
+
+  @SuppressWarnings(Array("org.wartremover.warts.Any"))
+  private val unimplemented: PartialFunction[Any, Unit] = {
```

Comment:
Why not `PartialFunction[Throwable, Unit]`?

Reply:
I tried that initially but unfortunately `matchResult` specifically asks for a ~`Receive`~ `PartialFunction[Any, Unit]` and changing this to `PartialFunction[Throwable, Unit]` makes it not compile. It's a bit weird now that I think about it, since I believe `PartialFunction` to be contravariant in the input type... :thinking: